gustaf/spline/create.py

```diff
@@ -0,0 +1,263 @@
+"""gustaf/spline/create.py
+
+Create operations for spline geometries
+"""
+
+
+import itertools
+
+import numpy as np
+from splinepy._spline import _RequiredProperties
+
+from gustaf import utils
+from gustaf import settings
+
+
+def extrude(spline, extrusion_vector=None):
+    """
+    Extrudes Splines
+
+    Parameters
+    ----------
+    spline: GustafSpline
+    extrusion_vector: np.ndarray
+    """
+    from gustaf.spline.base import GustafSpline
+
+    # Check input type
+    if not issubclass(type(spline), GustafSpline):
+        raise NotImplementedError("Extrude only works for splines")
+
+    # Check extrusion_vector
+    if extrusion_vector is not None:
+        # make flat extrusion_vector
+        extrusion_vector = np.asarray(extrusion_vector).ravel()
+    else:
+        raise ValueError("No extrusion extrusion_vector given")
+
+    # Check extrusion_vector dimension
+    # formulate correct cps
+    if spline.dim == extrusion_vector.shape[0]:
+        cps = spline.control_points
+    elif spline.dim < extrusion_vector.shape[0]:
+        expansion_dimension = extrusion_vector.shape[0] - spline.dim
+        # one smaller dim is allowed
+        # warn that we assume new dim is all zero
+        utils.log.warning(
+            f"Given extrusion vector is {expansion_dimension} dimension bigger than "
+            "spline's dim. Assuming 0.0 entries for new dimension.",
+        )
+        cps = np.hstack(
+            (
+                spline.control_points,
+                np.zeros((len(spline.control_points), expansion_dimension))
+            )
+        )
+    else:
+        raise ValueError(
+            "Dimension Mismatch between extrusion extrusion vector and spline."
+        )
+
+    # Start Extrusion
+    spline_dict = dict()
+
+    spline_dict["degrees"] = np.concatenate((spline.degrees, [1]))
+    spline_dict["control_points"] = np.vstack((cps, cps + extrusion_vector))
+    if "knot_vectors" in _RequiredProperties.of(spline):
+        spline_dict["knot_vectors"] = spline.knot_vectors + [[0, 0, 1, 1]]
+    if "weights" in _RequiredProperties.of(spline):
+        spline_dict["weights"] = np.concatenate(
+            (spline.weights, spline.weights))
+
+    return type(spline)(**spline_dict)
+
+
+def revolve(spline, axis=None, center=None, angle=None, n_knot_spans=None):
```

Comment:
The angle is expected to be radian? can we have default as degrees? It is just much friendlier